### PR TITLE
Add route validation

### PR DESCRIPTION
**Pull request by Aaron2000119**
Closes Gamify-IT/issues#308.
To test start the backend and test if you can access the routes without the cookie.

### CODE REVIEW

**Review by robojones — COMMENTED** [review]
Is there no test verifying that the cookie is actually validated? Or am I just not seeing them?

File: src/main/java/de/unistuttgart/crosswordbackend/controller/CrosswordController.java [hunk 1]
```diff
@@ -19,11 +18,19 @@
...
   QuestionMapper questionMapper;
 
   @GetMapping("/validate-crossword")
-  public boolean isValidCrosswordPuzzle(@RequestBody Set<QuestionDTO> questionDTOs) {
+  public boolean isValidCrosswordPuzzle(
+    @RequestBody final Set<QuestionDTO> questionDTOs,
+    @CookieValue("access_token") final String accessToken
+  ) {
+    jwtValidatorService.validateTokenOrThrow(accessToken);
+    jwtValidatorService.hasRolesOrThrow(accessToken, List.of("lecturer"));
```
[the remark this answers is not in the document]
Comment (Aaron2000119) [reply src/main/java/de/unistuttgart/crosswordbackend/controller/CrosswordController.java:33]: to validate if a crosswordpuzzle can be created out of the set of questions yes. This was ment for the lecturer interface to check, if you can create a crosswordpuzzle out of the questions which are submitted.

[the remark this answers is not in the document]
Comment (delvh) [reply src/main/java/de/unistuttgart/crosswordbackend/controller/CrosswordController.java:33]: 1. I don't think only a lecturer should be able to validate that
2. Even if you leave it in, you need to comment that!

**Comment by delvh** [comment]
Hmm, do we still merge today?

**Comment by Aaron2000119** [comment]
I would merge as soon as it is approved

File: src/main/java/de/unistuttgart/crosswordbackend/controller/CrosswordController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (madebyTimo) [reply src/main/java/de/unistuttgart/crosswordbackend/controller/CrosswordController.java:36]: I agree to Aaron. It is only needed at creation also only needed for lecturers

File: pom.xml [hunk 2]
```diff
@@ -49,7 +69,8 @@
 		<dependency>
 			<groupId>org.postgresql</groupId>
```
Comment (madebyTimo) [inline pom.xml:70]: The formattting of this file needs to be improved!

**Review by madebyTimo — APPROVED** [review]
weel done. Beside the formatting in the pom it is fine

**Pull request merged**